save.py

```diff
@@ -233,6 +234,84 @@ def rotating_streak(self) -> int:
     def character_streak(self) -> int:
         return get_latest_run(self.character, None).character_streak.streak
 
+    @property
+    def score(self) -> int:
+        return sum(bonus.score_bonus for bonus in self._get_score_bonuses())
+
+    @property
+    def score_breakdown(self) -> list[str]:
+        return [bonus.full_display for bonus in self._get_score_bonuses() 
+                        if bonus.should_show or bonus.score_bonus > 0]
```

Comment:
so we could return everything and have the client filter them based on the should_show or score = 0, or we can filter here. I don't have a strong opinion on either, I don't feel that we'll ever do anything with the ones that shouldn't show.